balancer/endpointsharding/endpointsharding.go

```diff
@@ -282,6 +291,15 @@ func (bw *balancerWrapper) UpdateState(state balancer.State) {
 	bw.es.mu.Lock()
 	bw.childState.State = state
 	bw.es.mu.Unlock()
+	// When a child balancer says it's IDLE, ping it to exit idle and reconnect.
+	// TODO: In the future, perhaps make this a knob in configuration.
+	if ei, ok := bw.Balancer.(balancer.ExitIdler); state.ConnectivityState == connectivity.Idle && ok {
+		go func() {
+			bw.es.childMu.Lock()
+			ei.ExitIdle()
```

Comment:
Don't we need to avoid this racing with Close?  Not worried about a deadlock or concurrency, but this sequence:

- Child reports IDLE
- This goroutine gets scheduled
- Parent calls `Close()`
- This goroutine runs and tells the child to `ExitIdle()` after `Close()`

Reply:
Added close guard.

Reply:
Good catch. I didn't even think of this constraint of the balancer.Balancer API of nothing called after Close(). So I guess ExitIdle even though a separate interface is the part of same API.

I did an grpcsync.Event, but let me know if you would rather have a bool.